### PR TITLE
Fix/exp overflow

### PR DESCRIPTION
**Pull request by BenjixD**
- Changed EXP to use long values
- Used log factorization to reduce computation size.

### CODE REVIEW

File: scripts/npc/2040043.js [hunk 1]
```diff
@@ -37,101 +37,101 @@ var status = 0;
...
+						cm.dispose();
+					}
+				}
+			} else {
+				cm.sendNext("Please tell your #bParty-Leader#k to come talk to me.");
+			}
+		}
+		
+		cm.dispose();
+	}
```
Comment (BenjixD) [inline scripts/npc/2040043.js:136]: spacing issues fixed for this file

File: src/client/MapleCharacter.java [hunk 2]
```diff
@@ -2479,77 +2479,93 @@ public void gainGachaExp(int gain) {
...
-        long total = Math.max(gain, -exp.get());
-        
-        if (level < getMaxLevel()) {
-            long leftover = 0;
-            long nextExp = exp.get() + total;
-            
-            if (nextExp > (long)Integer.MAX_VALUE) {
-                total = Integer.MAX_VALUE - exp.get();
-                leftover = nextExp - Integer.MAX_VALUE;
+    private void gainExpInternal(long gain, long equip, long party, boolean show, boolean inChat, boolean white) {
```
Comment (BenjixD) [inline src/client/MapleCharacter.java:2523]: re-wrote gainExpInternal

Comment (BenjixD) [inline src/client/MapleCharacter.java]: re-wrote gainExpInternal

File: src/client/MapleCharacter.java [hunk 3]
```diff
@@ -7205,7 +7218,17 @@ public void increaseEquipExp(int expGain) {
             }
 
             if ((nEquip.getItemLevel() < ServerConstants.USE_EQUIPMNT_LVLUP) || (itemName.contains("Reverse") && nEquip.getItemLevel() < 4) || (itemName.contains("Timeless") && nEquip.getItemLevel() < 6)) {
-                nEquip.gainItemExp(client, expGain);
+                // Apply Integer bounded increments of EXP
+                while(true) {
```
Comment (BenjixD) [inline src/client/MapleCharacter.java:7225]: updated to handle long values > MAX_INT

File: src/server/life/MapleMonster.java [hunk 4]
```diff
@@ -256,49 +256,40 @@ public boolean isAttackedBy(MapleCharacter chr) {
         return takenDamage.containsKey(chr.getId());
     }
 
-    private void distributeExperienceToParty(int pid, int exp, int killer, Map<Integer, Integer> expDist) {
+    private void distributeExperienceToParty(int pid, long exp, int killer, Map<Integer, Long> expDist) {
```
Comment (BenjixD) [inline src/server/life/MapleMonster.java:259]: updated distribute exp to handle long values using log computation

Comment (BenjixD) [inline src/server/life/MapleMonster.java]: updated distribute exp to handle long values using log computation

Comment (BenjixD) [inline src/server/life/MapleMonster.java]: updated distribute exp to handle long values using log computation.

Comment (BenjixD) [inline src/server/life/MapleMonster.java]: updated distribute exp to handle long values using log computation.

**Review by nansonzheng — APPROVED** [review]

**Pull request merged**